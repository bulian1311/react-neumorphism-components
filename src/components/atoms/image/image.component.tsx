import React, { FC } from 'react';
import PropTypes from 'prop-types';
import { StyledImage, StyledFigure } from './image.styled';
import { ImageProps } from './image.types';

export const Image: FC<ImageProps> = ({ src }) => {
  return (
    <StyledFigure src={src}>
      <StyledImage src={src} />
    </StyledFigure>
  );
};

Image.displayName = 'Image';

Image.defaultProps = {
  isFullWidth: false,
  isRounded: false,
};

Image.propTypes = {
  size: PropTypes.number,
  isFullWidth: PropTypes.bool,
  isRounded: PropTypes.bool,
};
