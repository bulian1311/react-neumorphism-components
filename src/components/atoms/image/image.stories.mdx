import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import * as stories from './image.stories.tsx';
import { Image } from './image.component.tsx';

<Meta title="Atoms/Image" />

# Image

Responsive images made easy. Images take time to load (if they load at all), for that is a good practice to
reserve the space before loading the image to avoid the page to jump around.

This Component also allow you to setup a fallback image in case the image its not loaded correctly,
try on the `Canvas` tab changing the src control to a unexisting image route and see what happens

## Props

<ArgsTable of={Image} />

## Example

<Canvas>
  <Story story={stories.Image} name="Image" />
</Canvas>
