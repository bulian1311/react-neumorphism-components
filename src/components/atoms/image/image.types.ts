export type ImageProps = {
  src: string;
  isRounded?: boolean;
  isFullWidth?: boolean;
  size?: number;
};
