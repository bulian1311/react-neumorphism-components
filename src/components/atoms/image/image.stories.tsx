import React from 'react';
import { Story } from '@storybook/react';

import { Image as ImageComponent } from './image.component';

const Template: Story = (args) => <ImageComponent {...args} src="" />;

export const Image = Template.bind({});

Image.args = {
  isRoundes: false,
  isFullWidth: false,
  size: 100
};

Image.argTypes = {
  isRoundes: {
    type: { name: 'boolean', required: false },
    name: 'Is Roundes',
    control: { type: 'boolean' },
  },
  isFullWidth: {
    type: { name: 'boolean', required: false },
    name: 'Is Roundes',
    control: { type: 'boolean' },
  },
  size: {
    type: { name: 'number', required: false },
    name: 'Size',
    control: { type: 'range', min: 0, max: 500, step: 1 },
  },
};
