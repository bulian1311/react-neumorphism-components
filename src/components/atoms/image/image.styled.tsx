import styled, { css } from 'styled-components';
import { ImageProps } from './image.types';

export const StyledImage = styled.img<ImageProps>`
  border-radius: ${({ isRounded }) => (isRounded ? '9999' : '5')}px;
`;

export const StyledFigure = styled.figure<ImageProps>`
  display: block;
  position: relative;
  width: ${({ size }) => (size ? size : '50')}px;
  height: ${({ size }) => (size ? size : '50')}px;

  ${({ isFullWidth }) => (isFullWidth ? 'width: 100%;' : '')}

  img {
    display: block;
    height: auto;
    width: 100%;
  }
`;
