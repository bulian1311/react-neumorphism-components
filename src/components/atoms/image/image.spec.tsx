import React from 'react';
import { Image } from './image.component';
import { render } from '@testing-library/react';

describe('Image', () => {
  it('matches snapshot', () => {
    const { container } = render(<Image src="" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
