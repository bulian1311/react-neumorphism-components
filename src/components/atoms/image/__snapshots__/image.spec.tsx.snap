// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches snapshot 1`] = `
<figure
  class="sc-gsDKAQ igVdxT"
  src=""
>
  <img
    class="sc-bdvvtL krNaSv"
    src=""
  />
</figure>
`;
